plugins/entando-plugin-jpdigital-exchange/src/test/java/org/entando/entando/web/digitalexchange/install/DigitalExchangeInstallResourceIntegrationTest.java

```diff
@@ -166,51 +179,137 @@ public void shouldInstallWidget() throws Exception {
 
         String componentCode = "de_test_widget";
 
-        installAndCheckForCompletion(componentCode);
+        try {
+            installAndCheckForCompletion(componentCode);
+
+            assertThat(widgetService.getWidget(componentCode)).isNotNull();
+
+
+        } finally {
+
+            try {
+                if (widgetService.getWidget(componentCode) != null) {
+                    widgetService.removeWidget(componentCode);
+                }
+
+            } catch (RestRourceNotFoundException ignored) {
```

Comment:
The `widgetService.getWidget(componentCode)` will never return null if the component is not found, instead it will always throw the `RestRourceNotFoundException`. On the other hand, the `removeWidget(componentCode)` method will throw a `ValidationGenericException` if the widget is not found. I think that you can remove the if and catch directly the `ValidationGenericException` instead.